qpmodel/ExprSubquery.cs

```diff
@@ -261,25 +261,28 @@ public override void Bind(BindContext context)
 
         public override Value ExecNonDistributed(ExecContext context, Row input)
         {
-            Debug.Assert(type_ != null);
-            Value expr = expr_().Exec(context, input);
+            lock(this)
```

Comment:
why this change? Do we already handled this in Exec()?

Reply:
The mainline code also fixes the bug. Keep the bug fix code in this commit the when merging code.
I will remove above code.